gui/src/main.ts

```diff
@@ -61,6 +61,31 @@ Vue.prototype.$hasAccessToCsm = function(role: string) {
   return false;
 };
 
+Vue.prototype.$getFeatureList = function() {
+  const unsupportedFeatures = this.$store.getters[
+    "userLogin/getUnsupportedFeatures"
+  ];
+  return unsupportedFeatures;
+};
+
+Vue.directive("feature", {
+  inserted(el, binding) {
+    const vueInstance: any = router.app.$root;
+    const unsupportedFeatures = vueInstance.$getFeatureList();
+    if (
+      unsupportedFeatures &&
+      unsupportedFeatures.unsupported_features &&
+      unsupportedFeatures.unsupported_features.find(

```

Comment:
If flattened object is used instead of array, then we will not have to find in array everytime we want to check if feature is supported. We can just check as below:
if (unsupportedFeatures && unsupportedFeatures[binding.value])

Reply:
Updated if condition as discussed.